staging/src/k8s.io/component-base/zpages/flagz/flagz.go

```diff
@@ -0,0 +1,170 @@
+}
+
+type Flagz struct{}
+
+type mux interface {
+	Handle(path string, handler http.Handler)
+}
+
+type templateData struct {
+	ComponentName string
+}
+
+func init() {
+	var err error
+	t := template.New("t").Funcs(funcMap)
+	tmpl, err = t.Parse(flagzTemplate)
+	if err != nil {
+		klog.Errorf("error while parsing gotemplate: %v", err)
+	}
+}
+
+func (f Flagz) Install(m mux, componentName string, flagSets []*cliflag.NamedFlagSets) {
+	flagzRegistry.installHandler(m, componentName, flagSets)
+}
+
+func (reg *Registry) installHandler(m mux, componentName string, flagSets []*cliflag.NamedFlagSets) {
+	m.Handle("/flagz",
+		metrics.InstrumentHandlerFunc("GET",
+			/* group = */ "",
+			/* version = */ "",
+			/* resource = */ "",
+			/* subresource = */ "/flagz",
+			/* scope = */ "",
+			/* component = */ "",
+			/* deprecated */ false,
+			/* removedRelease */ "",
+			reg.handleFlags(componentName, tmpl, flagSets)))
+}
+
+func (reg *Registry) handleFlags(componentName string, tmpl *template.Template, flagSets []*cliflag.NamedFlagSets) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		reg.once.Do(func() {
+			header, err := populateHeader(componentName, tmpl)
+			if err != nil {
+				klog.Errorf("error while populating flagz header: %v", err)
+				http.Error(w, "error while populating flagz header", http.StatusInternalServerError)
+				return
+			}
+
+			fmt.Fprint(&reg.response, header)
+			flags := sortedFlags(flagSets)
+
+			for _, flag := range flags {
+				if set, ok := flag.Annotations["classified"]; !ok || len(set) == 0 {
+					fmt.Fprint(&reg.response, flag.Name, "=", flag.Value, "\n")
+				} else {
+					fmt.Fprint(&reg.response, flag.Name, "=", "CLASSIFIED", "\n")
+				}
+			}
+		})
+		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+		var cachedResponse bytes.Buffer
+		_, err := cachedResponse.Write(reg.response.Bytes())
+		if err != nil {
+			klog.Errorf("error writing to cachedResponse: %v", err)
+			http.Error(w, "error writing to cachedResponse", http.StatusInternalServerError)
+			return
+		}
+		_, err = cachedResponse.WriteTo(w)
+		if err != nil {
+			klog.Errorf("error writing response: %v", err)
+			http.Error(w, "error writing response", http.StatusInternalServerError)
+		}
+	}
+}
+
+func populateHeader(componentName string, tmpl *template.Template) (string, error) {
+	if tmpl == nil {
+		return "", fmt.Errorf("received nil template")
+	}
+
+	data := templateData{
+		ComponentName: componentName,
+	}
+
+	var tpl bytes.Buffer
+	err := tmpl.Execute(&tpl, data)
+	if err != nil {
+		return "", fmt.Errorf("error populating flagz header: %w", err)
+	}
+
+	return tpl.String(), nil
+}
+
+func sortedFlags(flagSets []*cliflag.NamedFlagSets) []*pflag.Flag {
+	var flags []*pflag.Flag
+	for _, flagset := range flagSets {
+		for _, fs := range flagset.FlagSets {
+			fs.VisitAll(func(flag *pflag.Flag) {
+				if flag.Value != nil && flag.Value.String() != "" && flag.Value.String() != "[]" {
```

Comment:
it's surprising to filter out "empty" flags in a function that should be sorting

I also wouldn't omit empty flag values... they might have been explicitly unset from a default value

Reply:
> I also wouldn't omit empty flag values... they might have been explicitly unset from a default value

Makes sense, removed the handling of empty flags.